monkey/monkey_island/cc/ui/src/components/report-components/ransomware/AttackSection.tsx

```diff
@@ -76,39 +82,39 @@ function sortTelemetry(telemetry): void {
   });
 }
 
-function getLatestTelemetry(telemetry) {
-  let latestTelemetry = {};
-  for (let i = 0; i < telemetry.objects.length; i++) {
-    let monkey = telemetry.objects[i].monkey
+function getDataForTable(events, agents, machines): Array<TableRow> {
+  let tableData = [];
 
-    if (! (monkey in latestTelemetry)) {
-      latestTelemetry[monkey] = {};
+  for (let event of events) {
+    if (event['success'] === true) {
+      tableData.push({'hostname': getHostname(event['source'], agents, machines), 'file_path': event['file_path']['path']});
     }
-
-    telemetry.objects[i].data.files.forEach((file_encryption_telemetry) => {
-      latestTelemetry[monkey][file_encryption_telemetry.path] = file_encryption_telemetry.success
-    });
   }
 
-  return latestTelemetry;
+  return tableData;
 }
 
-function getDataForTable(telemetry): Array<TableRow> {
-  let tableData = [];
-
-  for (const monkey in telemetry) {
-    for (const path in telemetry[monkey]) {
-      if (telemetry[monkey][path]) {
-        tableData.push({'hostname': parseHostname(monkey), 'file_path': path});
+function getHostname(event_source, agents, machines): string {
+  let hostname = "unknown";
+
+  for (let agent of agents) {
+    if (event_source === agent['id']) {
+      for (let machine of machines) {
+        if (agent['machine_id'] === machine['id']) {
+          if ((machine['hostname'] !== null) && (machine['hostname'] !== '')) {
+            hostname = machine['hostname'];
+          }
+          else {
+            hostname = machine['network_interfaces'][0];
```

Comment:
This will include the subnet mask (e.g. `192.168.1.10/24`. I think we just need the IP.